### PR TITLE
[15.0][MIG] payment_bitcoin: migration to v15

### PR DESCRIPTION
**Pull request by dsolanki-initos**

### CODE REVIEW

File: payment_bitcoin/controllers/main.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        :param type: Accepted types: 'Codabar', 'Code11', 'Code128',
+        'EAN13', 'EAN8', 'Extended39', 'Extended93', 'FIM', 'I2of5', 'MSI',
+        'POSTNET', 'QR', 'Standard39', 'Standard93', 'UPCA', 'USPS_4State'
+        :param humanreadable: Accepted values: 0 (default) or 1.
+        1 will insert the readable value
+        at the bottom of the output image
+        """
+        try:
+            value = value.replace("$$", "?")
+            value = value.replace("*$", "&")
```
Comment (alihashaam) [inline payment_bitcoin/controllers/main.py]: combine both commands (line 59 and 60) in one

File: payment_bitcoin/i18n/de.po [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+"<div style=\"margin: 0px; padding: 0px;\">\n"
+"                <p style=\"margin: 0px; padding: 0px; font-size: 13px;\">\n"
+"                    Dear ,\n"
+"                    <br/>\n"
+"                    <br/>\n"
+"                    The Amount received for Order {{object.name}} is less than actual order amount.Please verify\n"
+"                </p>\n"
+"            </div>\n"
+"        "
+msgstr ""
```
Comment (alihashaam) [inline payment_bitcoin/i18n/de.po:35]: translation missing

File: payment_bitcoin/i18n/de.po [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+
+#. module: payment_bitcoin
+#: model_terms:ir.ui.view,arch_db:payment_bitcoin.view_account_config_min_unused_bitcoin
+msgid ""
+"<span class=\"o_form_label\">Hours</span>\n"
+"                            <span class=\"fa fa-lg fa-building-o\" aria-label=\"Addresses checking for orders which are not older than\" role=\"img\"/>"
+msgstr ""
```
Comment (alihashaam) [inline payment_bitcoin/i18n/de.po:51]: translation missing

File: payment_bitcoin/i18n/de.po [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+msgid ""
+"<span>\n"
+"                        <strong>In Minutes</strong>\n"
+"                    </span>"
+msgstr ""
+
+#. module: payment_bitcoin
+#: model:ir.model.fields,help:payment_bitcoin.field_payment_acquirer__deadline
+msgid ""
+"Add deadline to bitcoin payment, within this deadline payment should be done"
```
Comment (alihashaam) [inline payment_bitcoin/i18n/de.po]: translation missing

File: payment_bitcoin/i18n/de.po (hunk 3 above)
Comment (jans23) [reply payment_bitcoin/i18n/de.po:51]: For me "Addresses checking" sounds wrong. Perhaps "Address check" instead. German: "Adressenprüfung für Aufträge, die nicht älter sind als"

File: payment_bitcoin/models/payment_transaction.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+                transaction.duration = 0
+
+    @api.model_create_multi
+    def create(self, values_list):
+        # values['date'] = fields.Datetime.now()
+        for values in values_list:
+            if values.get("acquirer_id"):
+                acquirer = self.env["payment.acquirer"].browse(values["acquirer_id"])
+                if acquirer.provider == "bitcoin":
+                    values.get("reference")
```
Comment (alihashaam) [inline payment_bitcoin/models/payment_transaction.py]: is this needed?

File: payment_bitcoin/i18n/de.po (hunk 4 above)
Comment (jans23) [reply payment_bitcoin/i18n/de.po]: Better English would be: "Add a deadline to Bitcoin payments within which the payment should be made."

German: "Fügen Sie Bitcoin-Zahlungen eine Frist hinzu, innerhalb derer die Zahlung erfolgen soll."

File: payment_bitcoin/i18n/de.po (hunk 2 above)
Comment (jans23) [reply payment_bitcoin/i18n/de.po:35]: English requires a space behind the full stop.

German: "Der für den Auftrag {{objekt.name}} erhaltene Betrag ist geringer als der tatsächliche Auftragswert. Bitte überprüfen Sie dies."

File: payment_bitcoin/i18n/de.po (hunk 3 above)
Comment (jans23) [reply payment_bitcoin/i18n/de.po:51]: "Hours" translates to "Stunden"

File: payment_bitcoin/i18n/de.po (hunk 4 above)
Comment (jans23) [reply payment_bitcoin/i18n/de.po]: "In Minutes" translates to "In Minuten"

File: payment_bitcoin/i18n/de.po [hunk 6]
```diff
@@ -0,0 +1,396 @@
...
+msgstr "Adresse"
+
+#. module: payment_bitcoin
+#: model:ir.model.fields,field_description:payment_bitcoin.field_payment_transaction__bitcoin_address_link
+msgid "Address Link"
+msgstr "Adresslink"
+
+#. module: payment_bitcoin
+#: model_terms:ir.ui.view,arch_db:payment_bitcoin.view_account_config_min_unused_bitcoin
+msgid "Addresses checking for orders which are not older than"
```
Comment (jans23) [inline payment_bitcoin/i18n/de.po]: Better English: "Address check for orders which are not older than"

German: "Adressprüfung für Aufträge, die nicht älter sind als"

File: payment_bitcoin/i18n/de.po [hunk 7]
```diff
@@ -0,0 +1,396 @@
...
+msgstr ""
+
+#. module: payment_bitcoin
+#: model:ir.model.fields,help:payment_bitcoin.field_res_config_settings__bit_coin_order_older_than
+#: model:ir.model.fields,help:payment_bitcoin.field_webkul_website_addons__bit_coin_order_older_than
+#: model:ir.model.fields,help:payment_bitcoin.field_website_country_config__bit_coin_order_older_than
+msgid "Addresses checking for orders which are not older than."
```
Comment (jans23) [inline payment_bitcoin/i18n/de.po]: Better English: "Address check for orders which are not older than"

German: "Addresses checking for orders which are not older than"

Note that this string does contain a full stop but the previous string in line 71 doesn't. I think that should be made consistent.

**Comment by alihashaam** [comment]
@dsolanki-initos please do the small changes that I have mentioned

File: payment_bitcoin/i18n/de.po [hunk 8]
```diff
@@ -0,0 +1,396 @@
...
+"Bitcoin Address: <span><a target=\"_blank\"                     "
+"href=\"https://www.blockchain.com/btc/address/%(address_id)s?"
+"                    filter=5\">%(address_id1)s</a></span>, <span>%(rate)s "
+"</span> BTC"
+msgstr ""
+
+#. module: payment_bitcoin
+#: model:ir.actions.act_window,name:payment_bitcoin.action_bitcoin_address
+#: model:ir.ui.menu,name:payment_bitcoin.menu_bitcoin_address
+msgid "Bitcoin Addresses"
```
Comment (jans23) [inline payment_bitcoin/i18n/de.po:153]: German: "Bitcoin Adressen"

File: payment_bitcoin/i18n/de.po [hunk 9]
```diff
@@ -0,0 +1,396 @@
...
+#. module: payment_bitcoin
+#: model:ir.model.constraint,message:payment_bitcoin.constraint_bitcoin_address_name_uniq
+msgid "Bitcoin Address must be unique"
+msgstr "Bitcoin Adresse muss eindeutig sein"
+
+#. module: payment_bitcoin
+#: code:addons/payment_bitcoin/models/bitcoin.py:0
+#, python-format
+msgid ""
+"Bitcoin Address: <span><a target=\"_blank\"                     "
```
Comment (jans23) [inline payment_bitcoin/i18n/de.po:140]: German: "Bitcoin Adresse"

File: payment_bitcoin/i18n/de.po [hunk 10]
```diff
@@ -0,0 +1,396 @@
...
+
+#. module: payment_bitcoin
+#: model:ir.actions.act_window,name:payment_bitcoin.action_bitcoin_address
+#: model:ir.ui.menu,name:payment_bitcoin.menu_bitcoin_address
+msgid "Bitcoin Addresses"
+msgstr ""
+
+#. module: payment_bitcoin
+#: model_terms:ir.ui.view,arch_db:payment_bitcoin.view_account_config_min_unused_bitcoin
+msgid "Bitcoin Addresses checking"
```
Comment (jans23) [inline payment_bitcoin/i18n/de.po]: Better English: "Bitcoin Address check"

German: "Bitcoin Adressprüfung"

File: payment_bitcoin/i18n/de.po [hunk 11]
```diff
@@ -0,0 +1,396 @@
...
+msgstr "Bitcoin-Rate"
+
+#. module: payment_bitcoin
+#: model:ir.model,name:payment_bitcoin.model_bitcoin_rate_line
+msgid "Bitcoin Rate Lines"
+msgstr ""
+
+#. module: payment_bitcoin
+#: model:ir.model.fields,field_description:payment_bitcoin.field_bitcoin_rate__url
+msgid "Bitcoin Rate URL"
```
Comment (jans23) [inline payment_bitcoin/i18n/de.po:181]: German: "Bitcoin-Rate URL"

File: payment_bitcoin/i18n/de.po [hunk 12]
```diff
@@ -0,0 +1,396 @@
...
+"%(address)s with %(amount)s                                 BTC has been "
+"confirmed. It is missing                                 "
+"%(insufficiant_amount)s BTC."
+msgstr "Die Bitcoin-Transaktion %(transaction)s für %(address)s mit %(amount)s BTC wurde bestätigt. Es fehlen %(insufficiant_amount)s BTC."
+
+#. module: payment_bitcoin
+#: model:ir.actions.server,name:payment_bitcoin.ir_cron_bitcoin_payment_reconciliation_ir_actions_server
+#: model:ir.cron,cron_name:payment_bitcoin.ir_cron_bitcoin_payment_reconciliation
+#: model:ir.cron,name:payment_bitcoin.ir_cron_bitcoin_payment_reconciliation
+msgid "Bitcoin: Addresses Checking"
```
Comment (jans23) [inline payment_bitcoin/i18n/de.po]: Better English: "Bitcoin Address check"

German: "Bitcoin Adressprüfung"

File: payment_bitcoin/models/payment_transaction.py (hunk 5 above)
Comment (dsolanki-initos) [reply payment_bitcoin/models/payment_transaction.py]: No, will remove it

**Comment by dsolanki-initos** [comment]
@jans23, @alihashaam I've done changes as per the feedback.

**Comment by jans23** [comment]
@dsolanki-initos You didn't change the "addresses checking" term, didn't you?

**Comment by dsolanki-initos** [comment]
@jans23 Now check, I've changed before but I was testing in a different replica and it is behind so changes did not come in the previous commit.

**Comment by jans23** [comment]
I still see it [here](https://github.com/Nitrokey/odoo-modules/blob/15.0-mig-payment_bitcoin/payment_bitcoin/i18n/de.po#L50). I suggest to use "address check" instead of "addresses check".

**Comment by dsolanki-initos** [comment]
> I still see it [here](https://github.com/Nitrokey/odoo-modules/blob/15.0-mig-payment_bitcoin/payment_bitcoin/i18n/de.po#L50). I suggest to use "address check" instead of "addresses check".

Updated it

**Comment by alihashaam** [comment]
@jans23 

This MR was reviewed on Friday and can be merged

**Pull request merged**